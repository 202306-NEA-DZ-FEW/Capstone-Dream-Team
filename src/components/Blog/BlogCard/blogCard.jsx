import React from "react";

export default function BlogCard({ blog, index, language }) {
    return (
        <div className='dark:text-white lg:py-5 md:py-5 overflow-hidden transition-shadow duration-300 bg-white rounded'>
            <img
                src='https://images.pexels.com/photos/932638/pexels-photo-932638.jpeg?auto=compress&amp;cs=tinysrgb&amp;dpr=3&amp;h=750&amp;w=1260'
                className='object-cover w-full h-64 rounded'
                alt=''
            />
            <div className='py-2'>
                <p className='mb-2 text-xs font-semibold text-gray-600'>
                    {blog.publish_date}
                </p>

                <p className='text-2xl font-bold '>{blog.title}</p>
                <p className='mb-2 text-gray-700 line-clamp-3'>
                    {blog.content}
                </p>
            </div>
            <div className='flex items-center justify-between'>
                <a className='text-teal-500 inline-flex items-center md:mb-2 lg:mb-0 py-2'>
                    Learn More
                    <svg
                        class='w-4 h-4 ml-1'
                        viewBox='0 0 24 24'
                        stroke='currentColor'
                        stroke-width='2'
                        fill='none'
                        stroke-linecap='round'
                        stroke-linejoin='round'
                    >
                        <path d='M5 12h14'></path>
                        <path d='M12 5l7 7-7 7'></path>
                    </svg>
                </a>
                <span className='text-gray-600  mr-3 inline-flex items-center lg:ml-auto md:ml-0 ml-auto leading-none text-sm pr-1 py-2 md:mb-2 lg:mb-0'>
                    <svg
                        className='w-4 h-4 mr-1'
                        stroke='currentColor'
                        stroke-width='2'
                        fill='none'
                        stroke-linecap='round'
                        stroke-linejoin='round'
                        viewBox='0 0 24 24'
                    >
                        <path d='M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z'></path>
                        <circle cx='12' cy='12' r='3'></circle>
                    </svg>
                    {blog.views}
                </span>
            </div>
        </div>
    );
}
