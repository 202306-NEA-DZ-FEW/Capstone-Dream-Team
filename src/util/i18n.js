import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
