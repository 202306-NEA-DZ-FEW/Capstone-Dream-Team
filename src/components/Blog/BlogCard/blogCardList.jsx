import React from "react";

import BlogCard from "./blogCard";

export default function BlogCardList({ blogs, numToShow }) {
    const displayedBlogs = blogs.slice(0, numToShow);

    return (
        <div className='px-4 py-6 flex justify-center mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl md:px-4 lg:px-16'>
            <div className='grid gap-6 lg:grid-cols-3 sm:max-w-sm lg:max-w-full md:max-w-md mx-auto justify-center'>
                {displayedBlogs.map((blog, index) => (
                    <div key={index}>
                        <BlogCard blog={blog} />
                    </div>
                ))}
            </div>
        </div>
    );
}
