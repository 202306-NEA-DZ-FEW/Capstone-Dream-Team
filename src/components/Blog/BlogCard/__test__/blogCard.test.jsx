import { I18nextProvider } from "react-i18next";
import renderer from "react-test-renderer";

import BlogCard from "../blogCard";
import i18n from "../../../../util/i18n";
const blog = {
    id: "1",
    data: {
        title: "Sample Title 1",
        content: "Sample Content 1",
        publish_date: "2023-10-31",
    },
};

it("renders correctly", () => {
    const tree = renderer
        .create(
            <I18nextProvider i18n={i18n}>
                <BlogCard language='en' blog={blog} />
            </I18nextProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
