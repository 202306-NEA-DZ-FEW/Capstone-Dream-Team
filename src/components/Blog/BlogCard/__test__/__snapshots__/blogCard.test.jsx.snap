// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dark:text-white flex flex-col justify-center lg:py-5 md:py-5 overflow-hidden w-80  transition-shadow duration-300 rounded"
>
  <img
    alt="Sample Title 1"
    className="rounded-xl object-cover w-full h-56"
    src="https://images.pexels.com/photos/932638/pexels-photo-932638.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260"
  />
  <div
    className="py-2"
  >
    <p
      className="w-80 text-stone-900 text-xl font-medium capitalize line-clamp-2"
    >
      Sample Title 1
    </p>
  </div>
  <div
    className="flex justify-between items-center my-1"
  >
    <a
      className="text-teal-500 inline-flex items-center md:mb-2 lg:mb-0"
      href="/blogs/blog/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <button
        className="bg-orange-400 hover:bg-orange-600 active:bg-green-800 px-4 py-2 rounded-md text-white font-medium font-Outfit"
      />
    </a>
    <div
      className="flex"
    >
      <span
        className="text-gray-600 dark:text-white inline-flex items-center lg:ml-auto md:ml-0 ml-auto leading-none text-md"
      >
        2023-10-31
      </span>
    </div>
  </div>
</div>
`;
